Fails to build under Ruby 2.4.0.

`backup` has a firm requirement on `json` version `1.8.2`, which fails to build with Ruby 2.4.0 due to `Fixnum` and `Bignum` being merged into `Integer`.

```shell
current directory: /usr/local/lib/ruby/gems/2.4.0/gems/json-1.8.2/ext/json/ext/generator
make "DESTDIR="
compiling generator.c
generator.c: In function ‘generate_json’:
generator.c:867:25: error: ‘rb_cFixnum’ undeclared (first use in this function)
     } else if (klass == rb_cFixnum) {
                         ^
generator.c:867:25: note: each undeclared identifier is reported only once for each function it appears in
generator.c:869:25: error: ‘rb_cBignum’ undeclared (first use in this function)
     } else if (klass == rb_cBignum) {
```

Updating to `json` version `2.0.2` corrects this problem. Please update the `gemspec` when you get a chance, thanks!

--- a/.travis.yml
+++ b/.travis.yml
@@ -9,6 +9,7 @@ rvm:
   - 2.1.0
   - 2.2.0
   - 2.3.0
+  - 2.4.0
 matrix:
   include:
     - rvm: "2.3.0"
@@ -17,3 +18,4 @@ branches:
   only:
     - master
     - develop
+    - v4-stable

--- a/backup.gemspec
+++ b/backup.gemspec
@@ -29,7 +29,7 @@ Gem::Specification.new do |gem|
 
   gem.add_dependency "thor", "0.18.1"
   gem.add_dependency "open4", "1.3.0"
-  gem.add_dependency "fog", "1.28.0"
+  gem.add_dependency "fog", "~> 1.28"
   gem.add_dependency "excon", "~> 0.44"
   gem.add_dependency "unf", "0.1.3" # for fog/AWS
   gem.add_dependency "dropbox-sdk", "1.6.5"
@@ -41,11 +41,10 @@ Gem::Specification.new do |gem|
   gem.add_dependency "twitter", "5.5.0"
   gem.add_dependency "hipchat", "1.0.1"
   gem.add_dependency "flowdock", "0.4.0"
-  gem.add_dependency "json", "1.8.2"
   gem.add_dependency "dogapi", "1.11.0"
   gem.add_dependency "aws-ses", "0.5.0"
-  gem.add_dependency "qiniu", "6.5.1"
-  gem.add_dependency "nokogiri", "~> 1.6.0"
+  gem.add_dependency "qiniu", "~> 6.5"
+  gem.add_dependency "nokogiri", "~> 1.6"
 
   gem.add_development_dependency "rubocop", "0.45.0"
   gem.add_development_dependency "rake"

